### PR TITLE
Added SSH model

### PR DESCRIPTION
**Pull request by Anna-Mironova**
TODO:

- [x] Add GFLOPs and MParams

### CODE REVIEW

File: demos/python_demos/object_detection_demo_retinaface/detector.py [hunk 1]
```diff
@@ -25,14 +25,16 @@ def __init__(self, ie, model_path, face_prob_threshold, device='CPU'):
         model = ie.read_network(model_path, os.path.splitext(model_path)[0] + '.bin')
 
         assert len(model.input_info) == 1, "Expected 1 input blob"
-        assert len(model.outputs) == 12 or len(model.outputs) == 9, "Expected 12 or 9 output blobs"
+        expected_outputs_count = (6, 9, 12)
+        assert len(model.outputs) in expected_outputs_count, "Expected 12, 9 or 6 output blobs"
```
Comment (eaidova) [inline demos/python_demos/object_detection_demo_retinaface/detector.py:29]:
```suggestion
        assert len(model.outputs) in expected_outputs_count, "Expected {} output blobs".format(', '.join(expected_outputs_count)
```

**Review by eaidova — APPROVED** [review]

File: models/public/ssh/ssh.md [hunk 2]
```diff
@@ -0,0 +1,101 @@
+# ssh
```
Comment (IRDonch) [inline models/public/ssh/ssh.md:1]: Looks like it's a reimplementation of the original Caffe model, so it ought to be called `ssh-mxnet`.

Comment (Anna-Mironova) [reply models/public/ssh/ssh.md]: Renamed

File: demos/python_demos/common/models/retinaface.py [hunk 3]
```diff
@@ -19,20 +19,24 @@
...
 
 
 class RetinaFace(Model):
     def __init__(self, ie, model_path, threshold=0.5, mask_threshold=0.5):
         super().__init__(ie, model_path)
 
         assert len(self.net.input_info) == 1, "Expected 1 input blob"
-        assert len(self.net.outputs) == 12 or len(self.net.outputs) == 9, "Expected 12 or 9 output blobs"
+        expected_outputs_count = (6, 9, 12)
+        assert len(self.net.outputs) in expected_outputs_count, "Expected {} output blobs".format(
```
Comment (IRDonch) [inline demos/python_demos/common/models/retinaface.py:31]: This message ends up being "Expected 6, 9, 12 output blobs", which doesn't make sense. It should be "6, 9 or 12".

File: demos/python_demos/common/models/retinaface.py [hunk 4]
```diff
@@ -75,6 +80,8 @@ def __init__(self, detect_attributes=False):
             self._features_stride_fpn, [anchors.shape[0] for anchors in self._anchors_fpn.values()]
         ))
         self.landmark_std = 0.2 if detect_attributes else 1.0
+        self.nms_threshold = 0.5 if process_landmarks else 0.3
+        self.include_boundaries = False if process_landmarks else True
```
Comment (IRDonch) [inline demos/python_demos/common/models/retinaface.py:84]:
```suggestion
        self.include_boundaries = not process_landmarks
```

File: demos/python_demos/object_detection_demo/object_detection_demo.py [hunk 5]
```diff
@@ -171,10 +171,11 @@ def get_plugin_configs(device, num_streams, num_threads):
     return config_user_specified
 
 
-def draw_detections(frame, detections, palette, labels, threshold, draw_landmarks=False):
+def draw_detections(frame, detections, palette, labels, threshold):
     size = frame.shape[:2]
     for detection in detections:
         if detection.score > threshold:
+            draw_landmarks = isinstance(detection, DetectionWithLandmarks)
```
Comment (IRDonch) [inline demos/python_demos/object_detection_demo/object_detection_demo.py:178]: Seems like an unnecessary variable, IMO. You could just inline it in the `if` where it's used.

File: models/public/ssh-mxnet/model.yml [hunk 6]
```diff
@@ -0,0 +1,35 @@
...
+
+description: >-
+  SSH: Single Stage Headless Face Detector. More details provided in the repository
+  <https://github.com/deepinsight/mxnet-SSH> and paper <https://arxiv.org/abs/1708.03979>
+task_type: detection
+files:
+  - name: ssh-model-final.zip
+    size: 73316283
+    sha256: 1ac43c830fe8f5de069a35101cc38b03751c8502e9c62e9a945424a05e25dc6c
+    source: https://www.dropbox.com/s/06zmchmq4iwj6rt/ssh-model-final.zip?dl=1
```
Comment (IRDonch) [inline models/public/ssh-mxnet/model.yml:23]: Where does this link come from? I don't see any mentions of it in the repository.

File: models/public/ssh-mxnet/ssh-mxnet.md [hunk 7]
```diff
@@ -0,0 +1,101 @@
+# ssh-mxnet
+
+## Use Case and High-Level Description
+
+SSH: Single Stage Headless Face Detector. More details provided in the [repository](https://github.com/deepinsight/mxnet-SSH) and [paper](https://arxiv.org/abs/1708.03979)
```
Comment (IRDonch) [inline models/public/ssh-mxnet/ssh-mxnet.md:5]:
```suggestion
SSH: Single Stage Headless Face Detector. More details provided in the [repository](https://github.com/deepinsight/mxnet-SSH) and [paper](https://arxiv.org/abs/1708.03979).
```

**Review by IRDonch — CHANGES_REQUESTED** [review]

File: models/public/ssh-mxnet/model.yml (hunk 6 above)
Comment (Anna-Mironova) [reply models/public/ssh-mxnet/model.yml:23]: Link to the pre-trained model comes from https://github.com/deepinsight/insightface/wiki/Model-Zoo#1-face-detection-models

File: models/public/ssh-mxnet/ssh-mxnet.md (hunk 7 above)
Comment (Anna-Mironova) [reply models/public/ssh-mxnet/ssh-mxnet.md]: Done

File: demos/python_demos/object_detection_demo/object_detection_demo.py (hunk 5 above)
Comment (Anna-Mironova) [reply demos/python_demos/object_detection_demo/object_detection_demo.py]: Removed

File: demos/python_demos/common/models/retinaface.py (hunk 4 above)
Comment (Anna-Mironova) [reply demos/python_demos/common/models/retinaface.py]: Done

File: demos/python_demos/common/models/retinaface.py (hunk 3 above)
Comment (Anna-Mironova) [reply demos/python_demos/common/models/retinaface.py]: Done

File: demos/python_demos/common/models/retinaface.py (hunk 4 above)
Comment (IRDonch) [reply demos/python_demos/common/models/retinaface.py]: Come to think of it, if `include_boundaries` is just `not process_landmarks`, is it even necessary to store it in the class? Seems like it's only used in the `nms` call, which you could just edit to `nms(..., include_boundaries=not process_landmarks)`.

File: models/public/ssh-mxnet/model.yml (hunk 6 above)
Comment (IRDonch) [reply models/public/ssh-mxnet/model.yml:23]: OK. Weird that they would publish it like that.

File: demos/python_demos/common/models/retinaface.py (hunk 4 above)
Comment (Anna-Mironova) [reply demos/python_demos/common/models/retinaface.py]: `include_boundaries` variable has been removed, ` not process_landmarks` is used instead

**Review by IRDonch — APPROVED** [review]

**Pull request merged**